hw5_to_do_list/oop/js/base_classes.js

```diff
@@ -0,0 +1,49 @@
+class Component {
+    constructor(props) {
+        this.props = props
+        this.state = {}
+        this.children = props.children
+        const elementType = props.elementType || 'div'
+        this.element = document.createElement(elementType)
+        if (props.cssClass) {
+            this.element.className = props.cssClass
+        }
+    }
+
+    setState(state) {
+        this.state = { ...this.state, ...state }
+        this.update()
+    }
+
+    update() {
+        this.render()
+    }
+
+    updateDOM() { }
```

Comment:
👍 own lifecycle hook, nice!
I would call it something like `beforeRenderChildren`